src/GraphQl/Type/TypeBuilder.php

```diff
@@ -48,7 +48,7 @@ public function __construct(private readonly TypesContainerInterface $typesConta
     /**
      * {@inheritdoc}
      */
-    public function getResourceObjectType(?string $resourceClass, ResourceMetadataCollection $resourceMetadataCollection, Operation $operation, bool $input, bool $wrapped = false, int $depth = 0): GraphQLType
+    public function getResourceObjectType(?string $resourceClass, ResourceMetadataCollection $resourceMetadataCollection, Operation $operation, bool $input, bool $wrapped = false, int $depth = 0, bool $required = true): GraphQLType
```

Comment:
Not sure which other things from the `ApiProperty` we would need here?

Reply:
I'm trying to prevent having 10 arguments on one of our public interfaces, this method has already too many arguments the metadata classes are really helpful to condense information in one argument.

Reply:
That makes sense, let me update that